### PR TITLE
Fix bug in __getattr__ of NNCFNetwork

### PR DESCRIPTION
**Pull request by kshpv**
Add a capability to change the bound functions of a wrapped module by passing NNCFNetwork instance as a first argument

### CODE REVIEW

**Review by vshampor — CHANGES_REQUESTED** [review]
Use `pass` instead of `...`.

**Comment by kshpv** [comment]
> Use `pass` instead of `...`.

Done

**Review by vshampor — APPROVED** [review]

File: nncf/torch/nncf_network.py [hunk 1]
```diff
@@ -338,10 +338,36 @@ def insert_at_point(self, point: PTInsertionPoint, fn_list: List[Callable]):
...
 
     def __getattr__(self, name):
-        wrapped_module = super().__getattr__(MODEL_WRAPPED_BY_NNCF_ATTR_NAME)
-        if hasattr(wrapped_module, name):
-            return getattr(wrapped_module, name)
-        return super().__getattr__(name)
+        class NotFound:
+            pass
+
+        def get_NNCFNetwork_attr(self, name):
```
Comment (andrey-churkin) [inline nncf/torch/nncf_network.py:344]:
```suggestion
        def get_nncf_network_attr(self, name):
```

File: nncf/torch/nncf_network.py [hunk 2]
```diff
@@ -338,10 +338,36 @@ def insert_at_point(self, point: PTInsertionPoint, fn_list: List[Callable]):
             raise RuntimeError("Unsupported insertion type: {}".format(point.insertion_type))
 
     def __getattr__(self, name):
-        wrapped_module = super().__getattr__(MODEL_WRAPPED_BY_NNCF_ATTR_NAME)
-        if hasattr(wrapped_module, name):
-            return getattr(wrapped_module, name)
-        return super().__getattr__(name)
+        class NotFound:
```
Comment (andrey-churkin) [inline nncf/torch/nncf_network.py:341]: Why we can not use `None`?

```python
def get_nncf_network_attr(self, name):
    return self.__dict__.get(name)


attr = get_nncf_network_attr(self, name)

if attr is not None:
    return attr
```

Comment (kshpv) [reply nncf/torch/nncf_network.py:341]: If we would return None we can't distinguish cases when the attribute exists and its value is None and there is no such attribute.

File: nncf/torch/nncf_network.py (hunk 1 above)
Comment (kshpv) [reply nncf/torch/nncf_network.py]: Done

**Review by alexsu52 — APPROVED** [review]
Looks good to me

**Review by andrey-churkin — APPROVED** [review]

**Review by AlexanderDokuchaev — APPROVED** [review]

**Pull request merged**